rest/route/host.go

```diff
@@ -7,13 +7,72 @@ import (
 	"net/url"
 	"strconv"
 
+	"github.com/evergreen-ci/evergreen"
 	"github.com/evergreen-ci/evergreen/model/task"
 	"github.com/evergreen-ci/evergreen/rest/data"
 	"github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/evergreen-ci/evergreen/util"
 	"github.com/evergreen-ci/gimlet"
 	"github.com/pkg/errors"
 )
 
+////////////////////////////////////////////////////////////////////////
+//
+// PATCH /rest/v2/hosts/{host_id}
+
+type hostChangeStatusHandler struct {
+	Status string `json:"status"`
+	hostId string
+	sc     data.Connector
+}
+
+func makeChangeHostStatus(sc data.Connector) gimlet.RouteHandler {
+	return &hostChangeStatusHandler{
+		sc: sc,
+	}
+}
+
+func (h *hostChangeStatusHandler) Factory() gimlet.RouteHandler {
+	return &hostChangeStatusHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *hostChangeStatusHandler) Parse(ctx context.Context, r *http.Request) error {
+	h.hostId = gimlet.GetVars(r)["host_id"]
+	body := util.NewRequestReader(r)
+	defer body.Close()
+
+	if err := util.ReadJSONInto(body, h); err != nil {
+		return errors.Wrap(err, "Argument read error")
+	}
+
+	if !util.StringSliceContains(evergreen.ValidUserSetStatus, h.Status) {
+		return errors.New("Invalid host status")
+	}
+
+	return nil
+}
+
+func (h *hostChangeStatusHandler) Run(ctx context.Context) gimlet.Responder {
+	user := MustHaveUser(ctx)
+	foundHost, err := h.sc.FindHostById(h.hostId)
+	if err != nil {
+		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "Database error"))
+	}
+
+	if err := h.sc.SetHostStatus(foundHost, h.Status, user.Username()); err != nil {
+		return gimlet.MakeJSONErrorResponder(errors.Wrap(err, "Database error"))
+	}
+
+	host := &model.APIHost{}
+	if err = host.BuildFromService(*foundHost); err != nil {
```

Comment:
1. I think it's weird that we need to dereference this here. It certianly looks weird.
2. I suspect that this host status change in this route won't be reflected in this response.